pkg/placement/placement.go

```diff
@@ -72,220 +128,57 @@ func (p *Service) ReportDaprStatus(srv placementv1pb.Placement_ReportDaprStatusS
-				log.Debugf("Member is removed gracefully: %s", registeredMemberID)
+				log.Debugf("Stream connection is disconnected gracefully: %s", registeredMemberID)
+				p.membershipCh <- hostMemberChange{
+					cmdType: raft.MemberRemove,
+					host:    raft.DaprHostMember{Name: registeredMemberID},
+				}
 			} else {
-				log.Debugf("Member is removed with error: %s, %v", registeredMemberID, err)
+				// no actions for hashing table. Instead, MembershipChangeWorker will check
+				// host updatedAt and if now - updatedAt > faultyHostDetectMaxDuration, remove hosts.
+				log.Debugf("Stream connection is disconnected with the error: %v", err)
 			}
 
 			return nil
 		}
 	}
 }
 
-func (p *Service) addHost(ctx context.Context, srv placementv1pb.Placement_ReportDaprStatusServer) {
-	p.hostsLock.Lock()
-	defer p.hostsLock.Unlock()
-
-	p.hosts = append(p.hosts, srv)
-}
-
-// RemoveHost removes the host from the hosts list
-func (p *Service) RemoveHost(srv placementv1pb.Placement_ReportDaprStatusServer) {
-	for i := len(p.hosts) - 1; i >= 0; i-- {
-		if p.hosts[i] == srv {
-			p.hosts = append(p.hosts[:i], p.hosts[i+1:]...)
-		}
-	}
-}
-
-// PerformTablesUpdate updates the connected dapr runtimes using a 3 stage commit. first it locks so no further dapr can be taken
-// it then proceeds to update and then unlock once all runtimes have been updated
-func (p *Service) PerformTablesUpdate(hosts []placementv1pb.Placement_ReportDaprStatusServer,
-	options placementOptions) {
-	p.updateLock.Lock()
-	defer p.updateLock.Unlock()
-
-	if options.incrementGeneration {
-		p.generation++
-	}
-
-	o := placementv1pb.PlacementOrder{
-		Operation: "lock",
-	}
-
-	for _, host := range hosts {
-		err := host.Send(&o)
-		if err != nil {
-			log.Errorf("error updating host on lock operation: %s", err)
-			continue
-		}
-	}
-
-	v := fmt.Sprintf("%v", p.generation)
-
-	o.Operation = "update"
-	o.Tables = &placementv1pb.PlacementTables{
-		Version: v,
-		Entries: map[string]*placementv1pb.PlacementTable{},
-	}
-
-	for k, v := range p.entries {
-		hosts, sortedSet, loadMap, totalLoad := v.GetInternals()
-		table := placementv1pb.PlacementTable{
-			Hosts:     hosts,
-			SortedSet: sortedSet,
-			TotalLoad: totalLoad,
-			LoadMap:   make(map[string]*placementv1pb.Host),
-		}
-
-		for lk, lv := range loadMap {
-			h := placementv1pb.Host{
-				Name: lv.Name,
-				Load: lv.Load,
-				Port: lv.Port,
-				Id:   lv.AppID,
-			}
-			table.LoadMap[lk] = &h
-		}
-		o.Tables.Entries[k] = &table
-	}
-
-	for _, host := range hosts {
-		err := host.Send(&o)
-		if err != nil {
-			log.Errorf("error updating host on update operation: %s", err)
-			continue
-		}
-	}
-
-	o.Tables = nil
-	o.Operation = "unlock"
-
-	for _, host := range hosts {
-		err := host.Send(&o)
-		if err != nil {
-			log.Errorf("error updating host on unlock operation: %s", err)
-			continue
-		}
-	}
-}
-
-// ProcessRemovedHost removes a host from the hash table
-func (p *Service) ProcessRemovedHost(id string) {
```

Comment:
This is no longer required because raft FSM logic handles the hashing table updates.